File: csharp/armstrong-numbers/ArmstrongNumbers.cs
public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        string numberAsString = number.ToString();
        int power = numberAsString.Length;

        double sumOfPowers = numberAsString.Sum(digitChar =>
        {
           int digit = (int)Char.GetNumericValue(digitChar);
           return Math.Pow(digit, power); 
        });

        return number == (int)sumOfPowers;
    }
}

File: typescript/armstrong-numbers/armstrong-numbers.ts
export function isArmstrongNumber(number: number | bigint): boolean {
  const bigIntNumber = BigInt(number);
  const digits = String(number).split('');
  const power = BigInt(digits.length);

  const sum = digits.reduce((acc, digit) => {
    return acc + BigInt(digit) ** power;
  }, 0n);
  return sum === bigIntNumber;
}
